CMakeLists.txt

```diff
@@ -97,6 +97,9 @@ find_package(OpenMP REQUIRED)
 find_package(folly REQUIRED)
 include_directories(${folly_INCLUDE_DIRS})
 
+find_package(Microsoft.GSL REQUIRED)
```

Comment:
can we use boost span instead? https://live.boost.org/doc/libs/1_85_0/libs/core/doc/html/core/span.html